src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/ServiceBusTriggerBinding.cs

```diff
@@ -126,6 +106,7 @@ public async Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
             contract.Add("CorrelationId", typeof(string));
             contract.Add("UserProperties", typeof(IDictionary<string, object>));
             contract.Add("MessageReceiver", typeof(MessageReceiver));
+            contract.Add("ClientEntity", typeof(ClientEntity));
```

Comment:
Users will have to cast in their code to the correct derived type (Queue/Subscription)?

Reply:
Yes, it was the idea.